cassandra/cluster.py

```diff
@@ -1975,6 +1975,12 @@ def __init__(self, cluster, hosts, keyspace=None):
         while futures.not_done and not any(f.result() for f in futures.done):
             futures = wait_futures(futures.not_done, return_when=FIRST_COMPLETED)
 
+        if not any(f.result() for f in self._initial_connect_futures):
```

Comment:
I also questioned myself as well about that. But.. I don't see any good reason a user would try to connect to a cluster without any host. So, there is no real reason .connect() should succeed IMO.